tls/s2n_psk.c

```diff
@@ -153,6 +153,107 @@ S2N_CLEANUP_RESULT s2n_psk_parameters_wipe(struct s2n_psk_parameters *params)
     return S2N_RESULT_OK;
 }
 
+bool s2n_offered_psk_list_has_next(struct s2n_offered_psk_list *psk_list)
+{
+    return psk_list != NULL && s2n_stuffer_data_available(&psk_list->wire_data) > 0;
+}
+
+S2N_RESULT s2n_offered_psk_list_read_next(struct s2n_offered_psk_list *psk_list, struct s2n_offered_psk *psk)
+{
+    ENSURE_REF(psk_list);
+    ENSURE_MUT(psk);
+
+    uint16_t identity_size = 0;
+    GUARD_AS_RESULT(s2n_stuffer_read_uint16(&psk_list->wire_data, &identity_size));
+    ENSURE_GT(identity_size, 0);
+
+    uint8_t *identity_data = NULL;
+    identity_data = s2n_stuffer_raw_read(&psk_list->wire_data, identity_size);
+    ENSURE_REF(identity_data);
+
+    /**
+     *= https://tools.ietf.org/rfc/rfc8446#section-4.2.11
+     *# For identities established externally, an obfuscated_ticket_age of 0 SHOULD be
+     *# used, and servers MUST ignore the value.
+     */
```

Comment:
Doesn't this comment imply that we should be checking to see if the ticket age isn't 0 and erroring if that's the case?

Reply:
That's probably a good idea, since we can't accept resumption PSKs yet anyway.

Reply:
No, because the value being 0 is a SHOULD, not a MUST. Ignoring the value is a MUST though, which an error would violate.